src/Suggest/Suggest.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/*
+ * This file is part of the ONGR package.
+ *
+ * (c) NFQ Technologies UAB <info@nfq.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchDSL\Suggest;
+
+use ONGR\ElasticsearchDSL\BuilderInterface;
+use ONGR\ElasticsearchDSL\ParametersTrait;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+
+class Suggest implements BuilderInterface
+{
+    use ParametersTrait;
+
+    const TERM = 'term';
+    const COMPLETION = 'completion';
+    const PHRASE = 'phrase';
+    const CONTEXT = 'completion';
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var string
+     */
+    private $field;
+
+    /**
+     * @var string
+     */
+    private $text;
+
+    /**
+     * TermSuggest constructor.
+     * @param string $name
+     * @param string $field
+     * @param string $type
+     * @param string $text
+     * @param array $parameters
+     */
+    public function __construct($name, $field, $type, $text, $parameters = [])
```

Comment:
I would suggest to change order to: $name, $type, $text, $field as this order is defined in elasticsearch docs https://www.elastic.co/guide/en/elasticsearch/reference/current/search-suggesters.html